### PR TITLE
Move cloud functions into Observations, fix for docs and test

### LINKED ISSUE #3097: MAST: Tesscut doc needs updating, cloud API is no-op
As requested by @orionlee in https://github.com/astropy/astroquery/issues/1521#issuecomment-2365429961

Public documentation should probably be updated, as `TessCut` class still have the documented 
[`enable_cloud_dataset()`](https://astroquery.readthedocs.io/en/latest/api/astroquery.mast.TesscutClass.html#astroquery.mast.TesscutClass.enable_cloud_dataset) and [`disable_cloud_dataset()`](https://astroquery.readthedocs.io/en/latest/api/astroquery.mast.TesscutClass.html#astroquery.mast.TesscutClass.disable_cloud_dataset) methods that are essentially no-ops.

### PR DESCRIPTION
**Pull request by snbianco**
Closes #3097 

After talking with others at MAST, it looks like `enable_cloud_dataset` and `disable_cloud_dataset` are only applicable for `Observations` and no other classes (as of now). To minimize complexity, I decided to move the functions into `Observations` so as to not make them available to other classes that inherit from `MastQueryWithLogin`. This fixes the documentation issue as well.

There was also a problem with one of our tests that I fixed, so the test suite is passing now.

### CODE REVIEW

**Comment by snbianco** [comment]
Do the code coverage checks only use the tests in `test_mast.py`? Is that why they're failing?

File: astroquery/mast/core.py
Comment (bsipocz) [inline astroquery/mast/core.py:85]: I would say these should be deprecated rather than starting to break user code without notice; while these have no effect, the code do run with non-cloud returns, right?

The deprecations would work nicely if you keep overriding the methods in the classes that these work and not get the deprecation.

What do you think @pllim?

**Comment by bsipocz** [comment]
(minor note: could you also `mast` label these PRs at opening? Also, can add the next milestone, there should be only one open, for any of the PRs.
That would help with triaging.)

File: astroquery/mast/core.py
Comment (pllim) [reply astroquery/mast/core.py:85]: Yes, I think these should be deprecated first. It sounds like people might be using these in the wild whether intentionally or not.

**Comment by snbianco** [comment]
> (minor note: could you also `mast` label these PRs at opening? Also, can add the next milestone, there should be only one open, for any of the PRs. That would help with triaging.)

For sure, I'll remember this for future PRs!

File: astroquery/mast/core.py
Comment (snbianco) [reply astroquery/mast/core.py:85]: This makes sense! The latest version deprecates the functions in the base class.

Comment (pllim) [reply astroquery/mast/core.py:85]: Thanks! Hmm you could also use `@deprecated` decorator from `astropy.utils` but I'll leave that to astroquery maintainers to decide.

File: CHANGES.rst
Comment (pllim) [inline CHANGES.rst:215]:
```suggestion
- Deprecated ``enable_cloud_dataset`` and ``disable_cloud_dataset`` in classes where they
  are non-operational. They will be removed in a future release. [#3113]
```

File: astroquery/mast/core.py
Comment (snbianco) [reply astroquery/mast/core.py:85]: What's your preference @bsipocz? It looks like there are instances of both methods in the repo.

Comment (snbianco) [reply astroquery/mast/core.py:85]: Made that change! I followed the convention of the other deprecated functions and left the original docstrings in.

**Review by bsipocz — APPROVED** [review]
OK, this looks good. I'll rebase for merging though (please update the PR with a rebase instead of a merge commit next time).

**Pull request merged**
